File: app/src/main/java/info/horriblesubs/sher/adapter/ScheduleAdapter.java
package info.horriblesubs.sher.adapter;

import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import info.horriblesubs.sher.R;
import info.horriblesubs.sher.api.horrible.model.ScheduleItem;

public class ScheduleAdapter extends RecyclerView.Adapter<ScheduleAdapter.ViewHolder> {

    private final List<ScheduleItem> items;
    private final OnItemClick onItemClick;
    private final boolean isHome;
    private final int size;
    private boolean a;

    private ScheduleAdapter(OnItemClick itemClick, List<ScheduleItem> items, int size, boolean isHome) {
        this.onItemClick = itemClick;
        this.items = items;
        this.size = size;
        this.isHome = isHome;
        a = isHome;
    }

    @NotNull
    public static ScheduleAdapter get(OnItemClick itemClick, List<ScheduleItem> items) {
        return new ScheduleAdapter(itemClick, items, items == null ? 0 : items.size(), false);
    }

    @NotNull
    public static ScheduleAdapter home(OnItemClick itemClick, List<ScheduleItem> items) {
        return new ScheduleAdapter(itemClick, items, items == null ? 0 : items.size(), true);
    }

    @NotNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup group, int viewType) {
        return new ViewHolder(LayoutInflater.from(group.getContext()).inflate(R.layout.recycler_c, group, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, final int position) {
        if (a && isHome) holder.textView2.setText(items.get(position).getLeftTime());
        else holder.textView2.setText(items.get(position).getTime());
        holder.textView1.setText(Html.fromHtml(items.get(position).title));
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemClick.onItemClicked(items.get(position));
            }
        });
    }

    public void onToggle(boolean b) {
        a = b;
        notifyDataSetChanged();
    }

    @Override
    public int getItemViewType(int position) {
        return super.getItemViewType(position);
    }

    @Override
    public int getItemCount() {
        return size;
    }

    public interface OnItemClick {
        void onItemClicked(ScheduleItem item);
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        private final AppCompatTextView textView1, textView2;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            textView2 = itemView.findViewById(R.id.textView2);
            textView1 = itemView.findViewById(R.id.textView1);
        }
    }
}

File: app/src/main/java/info/horriblesubs/sher/ui/_extras/adapters/horrible/ScheduleAdapter.kt
package info.horriblesubs.sher.ui._extras.adapters.horrible

import android.view.ViewGroup
import androidx.core.text.parseAsHtml
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import info.horriblesubs.sher.R
import info.horriblesubs.sher.data.subsplease.api.model.ItemSchedule
import info.horriblesubs.sher.data.subsplease.api.model.imageUrl
import info.horriblesubs.sher.databinding.RecyclerItemDBinding
import info.horriblesubs.sher.ui._extras.listeners.OnItemClickListener
import info.horriblesubs.sher.ui.into
import info.horriblesubs.sher.ui.viewBindings

class ScheduleAdapter (
    val listener: OnItemClickListener<ItemSchedule.Show>,
    private val listItems: List<ItemSchedule.Show>
): RecyclerView.Adapter<ScheduleAdapter.Holder>() {

    override fun onCreateViewHolder(group: ViewGroup, viewType: Int) = Holder(group)

    override fun getItemCount(): Int = listItems.size

    override fun onBindViewHolder(holder: Holder, position: Int) {
        val t = listItems[position]
        holder.binding.root.setOnClickListener { listener.onItemClick(it, t, position) }
        Glide.with(holder.binding.imageView).load(t.imageUrl).transform().apply {
            transform(RoundedCorners(8), CenterCrop())
            placeholder(R.drawable.app_placeholder)
            timeout(30000)
        }.into(holder.binding.imageView, holder.binding.progressBar)
        holder.binding.root.setOnLongClickListener { view ->
            listener.onItemLongClick(view, t, position)
            true
        }
        holder.binding.timeTextView.text = if (t.time.isNotBlank()) t.time else "Never"
        holder.binding.titleTextView.text = t.title.parseAsHtml()
    }

    class Holder (
        group: ViewGroup,
        val binding: RecyclerItemDBinding = viewBindings(R.layout.recycler_item_d, group)
    ): RecyclerView.ViewHolder(binding.root)
}